services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/profiili/ProfiiliClientITest.kt

```diff
@@ -0,0 +1,316 @@
+import assertk.assertThat
+import assertk.assertions.hasClass
+import assertk.assertions.isEqualTo
+import assertk.assertions.isNotNull
+import assertk.assertions.isNull
+import assertk.assertions.messageContains
+import assertk.assertions.prop
+import com.fasterxml.jackson.module.kotlin.readValue
+import fi.hel.haitaton.hanke.OBJECT_MAPPER
+import fi.hel.haitaton.hanke.accessToken
+import fi.hel.haitaton.hanke.toJsonString
+import io.mockk.checkUnnecessaryStub
+import io.mockk.clearAllMocks
+import io.mockk.clearMocks
+import io.mockk.confirmVerified
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.verify
+import java.time.Instant
+import okhttp3.HttpUrl
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.context.SecurityContext
+import org.springframework.security.oauth2.core.OAuth2AccessToken
+import org.springframework.web.reactive.function.client.WebClient
+import org.springframework.web.reactive.function.client.WebClientResponseException
+
+class ProfiiliClientITest {
+
+    private lateinit var mockApiTokensApi: MockWebServer
+    private lateinit var mockGraphQl: MockWebServer
+    private lateinit var profiiliClient: ProfiiliClient
+    private val securityContext: SecurityContext = mockk()
+
+    @BeforeEach
+    fun setUp() {
+        clearAllMocks()
+
+        mockApiTokensApi = MockWebServer()
+        mockGraphQl = MockWebServer()
+
+        val apiTokensUrl = mockApiTokensApi.url("/api-tokens/").toString()
+        val graphQlUrl = mockGraphQl.url("/graphql/").toString()
+        val properties = ProfiiliProperties(graphQlUrl, apiTokensUrl, AUDIENCE)
+        profiiliClient = ProfiiliClient(properties, WebClient.builder())
+    }
+
+    @AfterEach
+    fun tearDown() {
+        mockApiTokensApi.shutdown()
+        mockGraphQl.shutdown()
+        checkUnnecessaryStub()
+        confirmVerified(securityContext)
+    }
+
+    @Nested
+    inner class GetVerifiedName {
+        @Test
+        fun `throws exception when accessToken not found from authentication`() {
+            every { securityContext.accessToken() } returns null
+
+            val failure = assertFailure { profiiliClient.getVerifiedName(securityContext) }
+
+            failure.all {
+                hasClass(VerifiedNameNotFound::class)
+                messageContains("User not authenticated")
+            }
+            assertThat(mockApiTokensApi.requestCount).isEqualTo(0)
+            verify { securityContext.accessToken() }
+        }
+
+        @Test
+        fun `throws exception when API tokens not found`() {
+            clearMocks(securityContext)
```

Comment:
Is `clearMocks` needed here when there is `clearAllMocks` in `setUp`?

Reply:
These were left over from some debugging I did earlier. Removed now.